semgrep/tests/e2e/test_semgrep_rules_repo.py

```diff
@@ -0,0 +1,22 @@
+import subprocess
+from pathlib import Path
+
+
+def test_semgrep_rules_repo(run_semgrep):
+    subprocess.check_output(
+        ["git", "clone", "--depth=1", "https://github.com/returntocorp/semgrep-rules"]
+    )
+    subprocess.check_output(["python", "-m", "semgrep", "--generate-config"])
+    subprocess.check_output(
+        [
+            "python",
+            "-m",
+            "semgrep",
+            "--dangerously-allow-arbitrary-code-execution-from-rules",
+            "--strict",
+            "--test",
+            "--test-ignore-todo",
+            "semgrep-rules",
+        ]
+    )
+    subprocess.check_output(["python", "-m", "semgrep", "--validate", "semgrep-rules"])
```

Comment:
I'm not fully sure what this does, @DrewDennison could you explain these commands and what they test?

Reply:
i'm pretty sure this test makes sure that the entire semgrep-rules repo has validating config files